devices/Mcp7940xx/Mcp7940m.cs

```diff
@@ -0,0 +1,599 @@
+        {
+            if (Alarm1IsEnabled() || Alarm2IsEnabled() || SquareWaveOutputIsEnabled())
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Checks if general purpose output is setting the state of the MFP pin to high.
+        /// </summary>
+        /// <returns>Returns <c>true</c> if MFP pin is high, <c>false</c> if not.</returns>
+        public bool GeneralPurposeOutputIsHigh()
+        {
+            return RegisterHelper.RegisterBitIsSet(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.GeneralPurposeOutput);
+        }
+
+        #endregion
+
+        #region Square Wave
+
+        /// <summary>
+        /// Enables square wave output from the MFP pin.
+        /// </summary>
+        public void EnableSquareWaveOutput()
+        {
+            RegisterHelper.SetRegisterBit(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.SquareWaveOutput);
+        }
+
+        /// <summary>
+        /// Disables square wave output from the MFP pin.
+        /// </summary>
+        public void DisableSquareWaveOutput()
+        {
+            RegisterHelper.ClearRegisterBit(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.SquareWaveOutput);
+        }
+
+        /// <summary>
+        /// Checks if square wave output is enabled.
+        /// </summary>
+        /// <returns>Returns <c>true</c> if square wave is enabled, <c>false</c> if not.</returns>
+        public bool SquareWaveOutputIsEnabled()
+        {
+            return RegisterHelper.RegisterBitIsSet(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.SquareWaveOutput);
+        }
+
+        /// <summary>
+        /// Gets the frequency of the square wave output on the MFP pin.
+        /// </summary>
+        /// <returns>A <see cref = "SquareWaveFrequency" /> object whose value is the current frequency of the square wave.</returns>
+        public SquareWaveFrequency GetSquareWaveFrequency()
+        {
+            byte control = RegisterHelper.ReadRegister(_I2cDevice, (byte)Register.Control);
+
+            return (SquareWaveFrequency)(control & (byte)ControlRegister.SquareWaveFrequencyMask);
+        }
+
+        /// <summary>
+        /// Sets the frequency of the square wave output on the MFP pin.
+        /// </summary>
+        /// <param name="frequency">The frequency of the square wave.</param>
+        public void SetSquareWaveFrequency(SquareWaveFrequency frequency)
+        {
+            RegisterHelper.MaskedSetRegisterBits(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.SquareWaveFrequencyMask, (byte)frequency);
+        }
+
+        #endregion
+
+        #region Clock
+
+        /// <summary>
+        /// Helper function to set the type of clock source being used.
+        /// </summary>
+        /// <param name="clockSource">The clocks oscillator configuration.</param>
+        internal void SetClockSource(ClockSource clockSource)
```

Comment:
Clock source just lets the chip know how its oscillator is soldered onto the board alongside the chip. I made it an internal function so I could write tests for it, but really, looking at it now, the logic can just be moved into the constructor. I'm not really sure if the corresponding GetClockSource function is even needed. Since the user probably will know their hardware layout, it might just be taking up space. Thoughts?